### PR TITLE
Introduces better diagnostic for overriding members with tuple arguments

### PR DESCRIPTION
**Pull request by Szer**
Fixes https://github.com/dotnet/fsharp/issues/15307

Any changes from maintainers to make this PR better are welcome!

### CODE REVIEW

**Comment by Szer** [comment]
@dotnet-policy-service agree

File: src/Compiler/Checking/CheckExpressions.fs [hunk 1]
```diff
@@ -11069,13 +11069,27 @@ and ApplyAbstractSlotInference (cenv: cenv) (envinner: TcEnv) (_: Val option) (a
...
+                              errorR(Error(FSComp.SR.tcOverrideArityMismatch details, memberId.idRange))
+                              []
+                         
+                         match slot with
+                         | FSMeth (_, _, valRef, _) ->
+                             match valRef.TauType with
+                             // https://github.com/dotnet/fsharp/issues/15307
+                             // check if abstract method expects tuple, give better error message
+                             | TType_fun(_,TType_fun(TType_tuple _,_,_),_) ->
+                                 if not slot.NumArgs.IsEmpty && slot.NumArgs.Head = 1 then
```
Comment (T-Gro) [inline src/Compiler/Checking/CheckExpressions.fs:11087]: Why is the `slot.NumArgs.Head = 1` here ;; can this scenario be covered in a test case as well please?

**Review by T-Gro — APPROVED** [review]

File: src/Compiler/FSComp.txt [hunk 2]
```diff
@@ -1694,3 +1694,4 @@ featureEscapeBracesInFormattableString,"Escapes curly braces before calling Form
 3564,parsMissingUnionCaseName,"Missing union case name"
 3565,parsExpectingType,"Expecting type"
 3566,tcMultipleRecdTypeChoice,"Multiple type matches were found:\n%s\nThe type '%s' was used. Due to the overlapping field names\n%s\nconsider using type annotations or change the order of open statements."
+3577,tcOverrideUsesMultipleArgumentsInsteadOfTuple,"This override takes a tuple instead of multiple arguments. Try to add an additional layer of parentheses at the call site, or remove parentheses at the abstract method declaration."
```
[the remark this answers is not in the document]
Comment (Happypig375) [reply src/Compiler/FSComp.txt:1697]: Can this error message depend on the actual source code, through formatting specifiers like the message directly above this one (The type '`%s`' was used)?

[the remark this answers is not in the document]
Comment (Szer) [reply src/Compiler/FSComp.txt]: I've looked into other examples with formatted string and couldn't find good example where we construct string from AST (like Fantomas is doing). Most of the formatted parameters are type names or variable names.

I thought about constructing example like
`member customThis.VerySpecialName((argName1: maybeArgTypeHere, argName2: definitelyArgTypeThere)` from given metadata, but decided to put simple and short examples of what to do instead. Reasons
- user code could have custom identifiers everywhere
- user code could have types explicitly stated or infered
- user code could be formatted with a lot of newlines
```fsharp
  member customThis.VerySpecialName(
    (argName1: maybeArgTypeHere, 
     argName2: definitelyArgTypeThere)
    ) = ()
```
- code which will deconstruct user code from AST could be very complex (ahem... Fantomas)
- user code could have multiple arguments, so error message could be very big

Suggestion to have an example of action is very helpful, I've added such examples in error code

[the remark this answers is not in the document]
Comment (0101) [reply src/Compiler/FSComp.txt]: I think the examples are good enough, we do the same in other error messages as well.

**Review by 0101 — APPROVED** [review]

**Pull request merged**